### ISSUE TITLE
CSV document indexing is not working properly when the id is a number

### ISSUE THREAD

**Issue by brunoocasali** [meilisearch/meilisearch#2723]
**Describe the bug**
Working on adding support in the Meilisearch Ruby SDK to v0.29, I found the following situation:

**To Reproduce**

When I try to add these documents using the CSV (`'Content-Type' => 'text/csv'`):

```ruby
# body encoded
"\"objectRef:number\",\"title:string\",\"comment:string\"\n\"1239\",\"Pride and Prejudice\",\"A great book\"\n\"4569\",\"Le Petit Prince\",\"A french book\"\n\"49\",\"Harry Potter and the Half-Blood Prince\",\"The best book\"\n"

# pretty printed:
"objectRef:number","title:string","comment:string"
"1239","Pride and Prejudice","A great book"
"4569","Le Petit Prince","A french book"
"49","Harry Potter and the Half-Blood Prince","The best book"
```

**Expected behavior**
Successfully index the documents.

**Current behavior**

I receive an error:

```ruby
"details"=>{"receivedDocuments"=>3, "indexedDocuments"=>0},
 "error"=>
  {"message"=>"Document identifier `1239.0` is invalid. A document identifier can be of type integer or string, only composed of alphanumeric characters (a-z A-Z 0-9), hyphens (-) and underscores (_).",
   "code"=>"invalid_document_id",
   "type"=>"invalid_request",
   "link"=>"https://docs.meilisearch.com/errors#invalid_document_id"},
 "duration"=>"PT0.009916S",
```

But my encoded body has the same structure as shown here: https://docs.meilisearch.com/learn/core_concepts/documents.html#csv

**Meilisearch version:** v0.29.0rc1, docker.

**Comment by curquiza** [comment]
Hello @brunoocasali 

Did you succeed in indexing your documents in v0.28.0?

If I'm correct, it should not have been possible since the document identifier contains a dot (`.`), or is considered a float, which should have been forbidden by the search engine:

```
A document identifier can be of type integer or string, only composed of alphanumeric characters (a-z A-Z 0-9), hyphens (-) and underscores (_).
```

**Comment by Kerollmops** [comment]
> If I'm correct, it should not have been possible since the document identifier contains a dot (.) [...]

It doesn't seem that the original CSV documents contain dots, it looks like the engine converted the ids (specified as numbers) and returned this error.

**Comment by Kerollmops** [comment]
Thank you @brunoocasali for the report, I fixed it on the milli side https://github.com/meilisearch/milli/pull/617 🎉

**Comment by brunoocasali** [comment]
> Thank you @brunoocasali for the report, I fixed it on the milli side [meilisearch/milli#617](https://github.com/meilisearch/milli/pull/617) 🎉

I saw the PR, sweet fix! 🥖 (tradition) haha

**Comment by curquiza** [comment]
Closed by #2732, will be in v0.29.0rc2
